versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/persist/ObjTypes.java

```diff
@@ -55,11 +60,23 @@ public static ObjType forShortName(@Nonnull String shortName) {
   private static final class Registry {
     private static final Map<String, ObjType> BY_ANY_NAME;
     private static final Set<ObjType> OBJ_TYPES;
+    private static final List<ObjTypeBundle.ObjTypeMapper> OBJ_TYPE_MAPPERS;
+
+    static ObjType maybeMapped(String name) {
+      for (ObjTypeBundle.ObjTypeMapper mapper : OBJ_TYPE_MAPPERS) {
+        ObjType type = mapper.mapGenericObjType(name);
```

Comment:
I'm not sure that this functionality should be part of the SPI. What if 2 or more bundles declare conflicting `ObjTypeMapper` instances? The result would be non-deterministic. There should imho be only one generic object type mapper, that would be invoked if all registered bundles failed to "understand" the object type.

Reply:
Yea - we probably will only ever need just one. Updated.

Reply:
Looking better now, but is it really necessary to load the (only) generic object mapper via service-loader? Why not just instantiate it normally?